main.py

```diff
@@ -1,574 +1,232 @@
+            logging.basicConfig(**logger_kwargs)
+        else:
+            logfile = log_path + '/sleepy_logs.log'
+            logging.basicConfig(filename=logfile, filemode='a+', **logger_kwargs)
+        #Load SleepyBaby
+        logging.info('Initializing...')
+        self.sleepy_baby = SleepyBaby(body_min_detection_confidence=body_min_detection_confidence, 
+                                      body_min_tracking_confidence=body_min_tracking_confidence,
+                                      face_min_detection_confidence=face_min_detection_confidence,
+                                      face_min_tracking_confidence=face_min_tracking_confidence)
+        if working_area:
+            self.sleepy_baby.set_working_area(working_area[0], working_area[1], working_area[2], working_area[3])
+        self.sleepy_baby.set_output(show_frame=show_frame, 
+                                    show_wrist_position=show_wrist_position, 
+                                    show_wrist_text=show_wrist_text, 
+                                    show_body_details=show_body_details, 
+                                    show_face_details=show_face_details,
+                                    show_progress_bar=show_progress_bar)
+        
+        logging.info('Initialization complete.')
+
+    def live(self, source:str = config['VIDEO_PATH'], return_image: bool = True):
+        """
+        Run App based on streaming video
+
+        Parameters
+        ----------
+        source : str
+            url of streaming video
+        return_image: bool
+            define if post processed image should be displayed, default: True
+        """        
+        self._process_streaming(source, apply_delay_between_frames=False, return_image=return_image)
+    
+    def recorded(self, source:str, return_image: bool = True):
+        """
+        Run App based on streaming video
+
+        Parameters
+        ----------
+        source : str
+            url of streaming video
+        return_image: bool
+            define if post processed image should be displayed, default: True
+        """        
+        self._process_streaming(source, apply_delay_between_frames=True, return_image=return_image)
+
+    def photo(self, source:str, output_size: tuple=None):
+        self.sleepy_baby.show_progress_bar = False
+        img = cv2.imread(source) 
+        output = self.sleepy_baby.processFrame(img)
+        if output_size:
+            output = cv2.resize(output, output_size)
+        cv2.imshow("Output", output)
+        cv2.waitKey()
+
+        
+    
+    def _process_streaming(self, source, apply_delay_between_frames=False, return_image=True):
+        try:
+            vcap = cv2.VideoCapture(source)
+            if vcap.isOpened():
+                success = True
+                logging.info("Start receiving frames.")
+                fps = vcap.get(cv2.CAP_PROP_FPS)
+                self.sleepy_baby.start_thread(frame_q, terminate_event)
```

Comment:
Ah, so for me `live` mode appears to be functional, however I haven't seen `recorded` mode work. I see you're continuing the pattern of using two threads (one push, one pop) for reading & processing images. I recall resorting to that for the `live` mode, but i don't recall that working for `recorded` clips, e.g. how I had it set up here:

https://github.com/calebolson123/BabySleepCoach/blob/master/main.py#L567

Have you seen `recorded` mode work w/ this two thread push/pop pattern? I do agree with aligning to one approach for both modes like you did here

Reply:
If you look at the code `live` and `recorded` calls the same function with the unique difference in the arguments population. In particular `recorded` sets to True the argumetn `apply_delay_between_frames` this is used to delay next frame based on the actual FPS of the video in order to reproduce it at standard speed.

Do you have troubles to use this command?

Reply:
Got it, I'll find some time to look into it further

Reply:
Had you have a chance to check it?